File: Algorithms/Greedy/coin.py
money = [500, 100, 50, 10, 1]
number = [1, 4, 8, 12, 50]
n = int(input())
for i in range(len(money)):
    cnt = 0
    while(n>=money[i] and cnt<number[i]):
        n-=money[i];
        cnt+=1
    print(f"{money[i]}->{cnt}")
    if(n==0):    break

File: Algorithms/Greedy/coin.cpp
#include<iostream>
using namespace std;

int main(){
    int n, cnt;
    int money[5]={500, 100, 50, 10, 1};
    int number[5]={1, 4, 8, 12, 50};
    cin>>n;
    for(int i=0;i<5;i++){
        cnt = 0;
        while(n>=money[i] && cnt<number[i]){
            n-=money[i];
            cnt++;
        }
        cout<<money[i]<<"->"<<cnt<<"\n";
        if (n==0)   break;
    }
    return 0;
}